core/publish/service.go

```diff
@@ -0,0 +1,146 @@
+package publish
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/anyproto/any-sync/app"
+	"github.com/anyproto/any-sync/commonfile/fileservice"
+	"github.com/anyproto/any-sync/util/crypto"
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/anyproto/anytype-heart/core/block"
+	"github.com/anyproto/anytype-heart/core/domain"
+	"github.com/anyproto/anytype-heart/core/filestorage/filesync"
+	"github.com/anyproto/anytype-heart/space"
+	"github.com/anyproto/anytype-heart/util/encode"
+)
+
+const CName = "common.core.publishservice"
+
+type PublishResult struct {
+	Cid string
+	Key string
+}
+type Service interface {
+	app.ComponentRunnable
+	Publish(ctx context.Context, spaceId, pageObjId string) (res PublishResult, err error)
+}
+
+type service struct {
+	commonFile      fileservice.FileService
+	fileSyncService filesync.FileSync
+	spaceService    space.Service
+	blockService    *block.Service
+	techSpaceId     string
+}
+
+func New() Service {
+	return new(service)
+}
+
+func (s *service) Init(a *app.App) error {
+	s.commonFile = app.MustComponent[fileservice.FileService](a)
+	s.fileSyncService = app.MustComponent[filesync.FileSync](a)
+	s.spaceService = app.MustComponent[space.Service](a)
+	s.blockService = a.MustComponent(block.CName).(*block.Service)
+	return nil
+}
+
+func (s *service) Run(_ context.Context) error {
+	s.techSpaceId = s.spaceService.TechSpaceId()
+	return nil
+}
+
+func (s *service) Close(_ context.Context) error {
+	return nil
+}
+
+func (s *service) Name() (name string) {
+	return CName
+}
+
+func (s *service) publishById(ctx context.Context, spaceId, pageObjId string) (res PublishResult, err error) {
```

Comment:
unused

Reply:
right, this one supposed to use id but not string from client. Removed for now in b5105a60a9e40fcce775c9f9ed20772919df115c